modules/terraform/output.go

```diff
@@ -49,6 +50,45 @@ func OutputRequiredE(t *testing.T, options *Options, key string) (string, error)
 	return out, nil
 }
 
+// OutputList calls terraform output for the given variable and return its value as a list.
+// If the output is a single value, then it returns a list with just one item.
+func OutputList(t *testing.T, options *Options, key string) []string {
+	out, err := OutputListE(t, options, key)
+	if err != nil {
+		t.Fatal(err)
+	}
+	return out
+}
+
+// OutputListE calls terraform output for the given variable and return its value as a list.
+// If the output is a single value, then it returns a list with just one item.
+func OutputListE(t *testing.T, options *Options, key string) ([]string, error) {
```

Comment:
Do we need to handle lists of ints? Or bools? Or nested structures, such as lists of maps or lists of lists?

Reply:
I'm not sure - formerly, I was just counting with strings. I can investigate _Terraform_ source code to catch some inspiration.

Reply:
According the [documentation](//www.terraform.io/docs/configuration/outputs.html#value), output type can be a `string`, a `list`, or a `map`. However, it doesn't state a precise value type for the later two. Checking the _Terraform_ [source code](//github.com/hashicorp/terraform/blob/5e511e1ea1ccbc6cfc1646d3dd3e88e8d81f3864/terraform/state.go#L917), the type is defined as:

* `string`
* `[]interface{}`
* `map[string]interface{}`
* `[]map[string]interface{}`

So theoretically, the list can be of other types than just `string`.

Maybe, let's keep it as `[]string` for the start and make it more generic on (later) demand? It can be quite complex to have it generic for complex structures.

For a `map`, I would say that a new function `OutputMap` would be good match.

Reply:
> Maybe, let's keep it as []string for the start and make it more generic on (later) demand? It can be quite complex to have it generic for complex structures.

Sounds good!